lib/matplotlib/testing/conftest.py

```diff
@@ -53,3 +53,25 @@ def mpl_test_settings(request):
             plt.switch_backend(prev_backend)
         _do_cleanup(original_units_registry,
                     original_settings)
+
+
+@pytest.fixture
+def mpl_image_comparison_parameters(request, extension):
+    # This fixture is applied automatically by the image_comparison decorator.
+    #
+    # The sole purpose of this fixture is to provide an indirect method of
+    # obtaining parameters *without* modifying the decorated function
+    # signature. In this way, the function signature can stay the same and
+    # pytest won't get confused.
+    # We annotate the decorated function with any parameters captured by this
+    # fixture so that they can be used by the wrapper in image_comparison.
+    baseline_images = request.keywords['baseline_images'].args[0]
```

Comment:
Looks like from a quick skim through the codebase that baseline_images is always a list with a single element.
1. Should we switch to having a single `baseline_image` (singular)?  Perhaps the ability to compare multiple images is useful, though.
2. If we want to assert that baseline_images is always of length 1, I prefer
```
baseline_image, = request.keywords["baseline_images"].args
```
which will error when multiple values are passed.
If we don't, then we shouldn't silently drop extra values here.